src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileValidationServiceImpl.java

```diff
@@ -3,28 +3,169 @@
+                                  fileTotalApprovedAmountsMatchHeaderTotalApprovedAmount(requestExtractFile));
+        LOG.info("Header validation : " + ((headerValidationPassed) ? "PASSED" : "FAILED"));
+        return headerValidationPassed;
+    }
+
+    public void performRequestDetailLineValidation(ConcurRequestExtractRequestDetailFileLine detailFileLine) {
+        if (requestDetailFileLineIsCashAdvance(detailFileLine)) {
+            boolean lineValidationPassed = false;
+            lineValidationPassed = (requestedCashAdvanceHasNotBeenUsedInExpenseReport(detailFileLine) &&
+                                    requestedCashAdvanceIsNotBeingDuplicated(detailFileLine) &&
+                                    requestedCashAdvanceAccountingInformationIsValid(detailFileLine));
+            detailFileLine.getValidationResult().setValidCashAdvanceLine(lineValidationPassed);
+            LOG.info("Detail File Line validation : " + ((lineValidationPassed) ? "PASSED" : "FAILED"));
+            LOG.info(KFSConstants.NEWLINE + ((lineValidationPassed) ? KFSConstants.EMPTY_STRING : detailFileLine.toString()));
+        }
+    }
+
+    /**
+     * A request detail line is considered a cash advance when request entry detail type is set to the cash advance indicator.
+     *
+     * @param detailFileLine
+     * @return boolean
+     */
+    private boolean requestDetailFileLineIsCashAdvance(ConcurRequestExtractRequestDetailFileLine detailFileLine) {
+        if (StringUtils.isNotEmpty(detailFileLine.getRequestEntryExpenseType()) &&
+            detailFileLine.getRequestEntryExpenseType().equalsIgnoreCase(ConcurRequestExtractPdpConstants.ValidationConstants.CASH_ADVANCE_INDICATOR)) {
+            detailFileLine.getValidationResult().setCashAdvanceLine(true);
         }
         else {
-            ConcurRequestExtractFile requestExtractFile = requestExtractFiles.get(0);
-            LOG.info("batchDate=" + requestExtractFile.getBatchDate() + "=   recordCount=" + requestExtractFile.getRecordCount() + "=     totalApprovedAmount=" + requestExtractFile.getTotalApprovedAmount() + "=");
-            headerValidationPassed = (fileRowCountMatchesHeaderRowCount(requestExtractFile) && fileApprovedAmountsMatchHeaderApprovedAmount(requestExtractFile));
+            detailFileLine.getValidationResult().setCashAdvanceLine(false);
+            detailFileLine.getValidationResult().addMessage(new String(ConcurRequestExtractPdpConstants.ValidationErrorMessages.NOT_CASH_ADVANCE_DATA_LINE));
+        }
+        return detailFileLine.getValidationResult().isCashAdvanceLine();
+    }
+
+    /**
+     * If a cash advance has been used in an expense report, the expense report id will have information in it.
+     * Bypass that transaction.
+     *
+     * @param detailFileLine
+     * @return boolean
+     */
+    private boolean requestedCashAdvanceHasNotBeenUsedInExpenseReport(ConcurRequestExtractRequestDetailFileLine detailFileLine) {
```

Comment:
this method name is misleading as this method does not just perform validation and returns a boolean, it is also changing the input object.